in_toto/models/gpg.py

```diff
@@ -0,0 +1,250 @@
+    passed homedir.
+
+    The executed base command is defined in
+    securesystemslib.gpg.constants.GPG_SIGN_COMMAND.
+
+    Arguments:
+      payload: The bytes to be signed.
+    Raises:
+      securesystemslib.exceptions.FormatError:
+        If the keyid was passed and does not match
+        securesystemslib.formats.KEYID_SCHEMA.
+      ValueError: the gpg command failed to create a valid signature.
+      OSError: the gpg command is not present or non-executable.
+      securesystemslib.exceptions.UnsupportedLibraryError: the gpg command is
+        not available, or the cryptography library is not installed.
+      securesystemslib.gpg.exceptions.CommandError: the gpg command returned a
+        non-zero exit code.
+      securesystemslib.gpg.exceptions.KeyNotFoundError: the used gpg version is
+        not fully supported and no public key can be found for short keyid.
+      Returns:
+        Returns a ``_LegacyGPGSignature`` class instance.
+    """
+
+    sig_dict = gpg.create_signature(payload, self.keyid, self.homedir)
+    return _LegacyGPGSignature(**sig_dict)
+
+
+@dataclass
+class _LegacyGPGKey(Key):
+  """A container class representing public key portion of a GPG key.
+  Provides a verify method to verify a cryptographic signature with a
+  gpg-style rsa, dsa or ecdsa public key on the instance.
+
+  Attributes:
+    type: Key type, e.g. "rsa", "dsa" or "ecdsa".
+    method: GPG Key Scheme, For example:
+      "pgp+rsa-pkcsv1.5", "pgp+dsa-fips-180-2", and "pgp+eddsa-ed25519".
+    hashes: list of GPG Hash Algorithms, e.g. "pgp+SHA2".
+    keyval: Opaque key content.
+    keyid: Key identifier that is unique within the metadata it is used in.
+      Keyid is not verified to be the hash of a specific representation
+      of the key.
+    creation_time: Unix timestamp when GPG key was created.
+    validity_period: Validity of the GPG Keys in days.
+    subkeys: A dictionary containing keyid and GPG subkey.
+  """
+
+  type: str
+  method: str
+  hashes: List[str]
+  keyval: Dict[str, str]
+  keyid: str
+  creation_time: Optional[int] = None
+  validity_period: Optional[int] = None
+  subkeys: Optional[Dict[str, "_LegacyGPGKey"]] = None
+
+  @classmethod
+  def from_dict(cls, keyid: str, key_dict: Dict[str, Any]):
+    """Creates ``_LegacyGPGKey`` object from its json/dict representation.
+    Raises:
+      KeyError, TypeError: Invalid arguments.
+    """
+    subkeys_dict = key_dict.get("subkeys")
+
+    gpg_subkeys = None
+    if subkeys_dict:
+      gpg_subkeys = {
+        _keyid: _LegacyGPGKey.from_dict(_keyid, subkey_dict)
+        for (_keyid, subkey_dict) in subkeys_dict.items()
+      }
+
+    return cls(
+      key_dict["type"],
+      key_dict["method"],
+      key_dict["hashes"],
+      key_dict["keyval"],
+      keyid,
+      key_dict.get("creation_time"),
+      key_dict.get("validity_period"),
+      gpg_subkeys,
+    )
+
+  @classmethod
+  def from_legacy_dict(cls, key_dict: Dict[str, Any]):
```

Comment:
I don't think this class needs separate `from_dict` and `from_legacy_dict`

Reply:
`from_dict` is an derived method and pylint throws `arguments-differ` if I remove `keyid` argument from it.

Reply:
Okay, makes sense. `_LegacyGPGKey.from_legacy_dict` will be the pendant to `SSlibKey.from_securesystemslib_key`.